test/test_monitoring.py

```diff
@@ -439,6 +439,9 @@ def test_not_master_error(self):
             client.pymongo_test.test.find_one_and_delete({})
         except NotMasterError as exc:
             error = exc.errors
+            import traceback
+            self.assertIn("full error", str(exc))
+            self.assertIn("full error", traceback.format_exc())
```

Comment:
This is exactly what I had in mind. What do you think about moving these assertions to their own test cases (maybe to a new file `test_errors.py`)? You can use the error classes directly, like:
```
exc = NotMasterError(...)
self.assertIn("full error", str(exc))
try:
    raise exc
except NotMasterError:
    self.assertIn("full error", traceback.format_exc())
```